common/include/pcl/common/fps_counter.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2014-, Open Perception Inc.
+ *
+ *  All rights reserved
+ */
+
+#pragma once
+
+#include <pcl/common/time.h>
+
+#include <functional>
+
+namespace pcl {
+
+class FpsCounter {
+public:
+  FpsCounter(unsigned counter_type = 0, unsigned reset_iter = 10)
+  : counter_type_(counter_type)
+  , reset_iter_(reset_iter)
+  , count_(0)
+  , start_time_(now())
+  , duration_(0.0)
+  , last_reset_(now())
+  {}
+
+  unsigned counter_type_;
+  unsigned reset_iter_;
+  unsigned count_;
+  double start_time_;
+  double duration_;
+  double last_reset_;
+
+  unsigned
+  getCount()
+  {
+    return count_;
+  }; // returns the number of frames counted till now
+  void
+  tick(); // add a frame
+  double
+  now()
+  {
+    return pcl::getTime();
+  }; // appropriate return for the time
+};
+
+class FpsCounterConditional : public FpsCounter {
+public:
+  FpsCounterConditional(auto fps_calc,
+                        auto reset_condition,
+                        auto print_details,
+                        auto reset,
+                        auto update_duration)
+  : fps_calc(fps_calc)
+  , reset_condition(reset_condition)
+  , print_details(print_details)
+  , reset(reset)
+  , update_duration(update_duration)
+  {}
+
+  void
+  tick()
+  {
+
+    double time_now = now();
+    double fps = fps_calc(time_now);
```

Comment:
You're offloading the core task of the class to the user: FPS Counting. This is either:
* bad naming or
* bad design

Having to implement simple stuff again and again makes an API bad. I'd advise reducing the flexibility. Even if it doesn't cover all use cases, it'd shine some light on where the constrained design lacks.

Reply:
Yes you are right. This is an oversight on the fps calculation since before it was only a printing. 
Do you think adding a get method (with a fps member) as done for the count in the base class would be enough to avoid the offloading?

Reply:
@kunaltyagi do you think the new structure solves the offloading? I made all the lambdas as methods in the derived class while leaving only the printing outside since it needs a capture to the `stop_computing`.